lib/dtutils.lua

```diff
@@ -0,0 +1,866 @@
+
+  RETURN VALUE
+    result - the incremented filename
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.filename_increment(filepath)
+
+  -- break up the filepath into parts
+  local path = dtutils.get_path(filepath)
+  local basename = dtutils.get_basename(filepath)
+  local filetype = dtutils.get_filetype(filepath)
+
+  -- check to see if we've incremented before
+  local increment = string.match(basename, "_(%d-)$")
+
+  if increment then
+    -- we do 2 digit increments so make sure we didn't grab part of the filename
+    if string.len(increment) > 2 then
+      -- we got the filename so set the increment to 01
+      increment = "01"
+    else
+      increment = string.format("%02d", tonumber(increment) + 1)
+      basename = string.gsub(basename, "_(%d-)$", "")
+    end
+  else
+    increment = "01"
+  end
+  local incremented_filepath = path .. basename .. "_" .. increment .. "." .. filetype
+
+  dt.print_error("original file was " .. filepath)
+  dt.print_error("incremented file is " .. incremented_filepath)
+
+  return incremented_filepath
+end
+
+--[[
+  NAME
+    dutils.groupIfNotMember - group a new file with an existing one
+
+  SYNOPSIS
+    result = dtutils.groupIfNotMember(image, new_image)
+      image - an existing image in the darktable database
+      new_image - the new image to group with the existing image
+
+  DESCRIPTION
+    groupIfNotMember checks a file to see if it's a member of the group
+    with an existing image.  If not, then the group leader of the existing 
+    image is found and the new image is grouped with it.
+
+  RETURN VALUE
+    none
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.groupIfNotMember(img, new_img)
+  local image_table = img:get_group_members()
+  local is_member = false
+  for _,image in ipairs(image_table) do
+    dt.print_error(image.filename .. " is a member")
+    if image.filename == new_img.filename then
+      is_member = true
+      dt.print_error("Already in group")
+    end
+  end
+  if not is_member then
+    dt.print_error("group leader is "..img.group_leader.filename)
+    new_img:group_with(img.group_leader)
+    dt.print_error("Added to group")
+  end
+end
+
+--[[
+  NAME
+    dutils.sanitize_filename - take care of spaces in filenames
+
+  SYNOPSIS
+    result = dtutils.sanitize_filename(filepath)
+      filepath - an optional path and filename with spaces
+
+  DESCRIPTION
+    sanitize_filename escapes spaces in filenames so that the can be passed as arguments
+
+  RETURN VALUE
+    result - sanitized filename
+
+  ERRORS
+    Only spaces in filenames are sanitized, but not in the path
+
+  
+]]
+
+function dtutils.sanitize_filename(filepath)
+  local path = dtutils.get_path(filepath)
+  local basename = dtutils.get_basename(filepath)
+  local filetype = dtutils.get_filetype(filepath)
+
+  local sanitized = string.gsub(basename, " ", "\\ ")
+
+  return path .. sanitized .. "." .. filetype
+end
+
+--[[
+  NAME
+    dutils.show_status - show the status while exporting images
```

Comment:
I'm not sure where to put this either.  If we move it to libPlugin and you write a contrib script do you want to include libPlugin? It wouldn't hurt anything.  I'll think on it...